manager.go

```diff
@@ -216,6 +218,12 @@ func (m *Manager) SubsystemUpdates(ctx context.Context) {
 		if err != nil {
 			m.logger.Warn(err)
 		}
+		if !reflect.DeepEqual(m.cfg.AdvancedSettings.ViamServerExtraEnvVars, m.prevCfg.AdvancedSettings.ViamServerExtraEnvVars) {
+			m.logger.Infow("Detected changed environment variables. Restarting viam-server at next opportunity.",
+				"current", m.cfg.AdvancedSettings.ViamServerExtraEnvVars,
+				"previous", m.prevCfg.AdvancedSettings.ViamServerExtraEnvVars)
+			needRestart = true
+		}
```

Comment:
somewhat minor, but is there a reason to have the needs restart logic here instead of inside the viamserver.Update method?